src/main/java/mate/academy/springboot/datajpa/service/impl/CategoryServiceImpl.java

```diff
@@ -0,0 +1,37 @@
+package mate.academy.springboot.datajpa.service.impl;
+
+import mate.academy.springboot.datajpa.model.Category;
+import mate.academy.springboot.datajpa.repository.CategoryRepository;
+import mate.academy.springboot.datajpa.service.CategoryService;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CategoryServiceImpl implements CategoryService {
+    private final CategoryRepository categoryRepository;
+
+    public CategoryServiceImpl(CategoryRepository categoryRepository) {
+        this.categoryRepository = categoryRepository;
+    }
+
+    @Override
+    public Category save(Category category) {
+        return categoryRepository.save(category);
+    }
+
+    @Override
+    public Category getById(Long id) {
+        return categoryRepository.getById(id);
+    }
+
+    @Override
+    public Category deleteById(Long id) {
+        Category deletedCategory = categoryRepository.getById(id);
+        categoryRepository.deleteById(id);
+        return deletedCategory;
+    }
```

Comment:
let's not make additional request to get deleted category, just return void
it's fine in this case